src/stdx/unshare.zig

```diff
@@ -0,0 +1,262 @@
+        }
+    }
+}
+
+/// Implementation of `ip link` equivalent to
+///
+/// ```
+/// ip link set up dev lo
+/// ```
+///
+/// This brings up the loopback device so that networking
+/// over 127.0.0.1 works.
+pub fn linux_ip_link_loopback() !void {
+    assert(builtin.os.tag == .linux);
+
+    // Open a netlink socket with the NETLINK.ROUTE protocol.
+    const sock = std.posix.socket(
+        linux.AF.NETLINK,
+        std.posix.SOCK.RAW,
+        linux.NETLINK.ROUTE,
+    ) catch |err| {
+        log.err("failed to create netlink socket: {}", .{err});
+        return error.IpLink;
+    };
+    defer std.posix.close(sock);
+
+    const addr = linux.sockaddr.nl{
+        .family = linux.AF.NETLINK,
+        .pid = 0,
+        .groups = 0,
+    };
+    std.posix.bind(sock, @ptrCast(&addr), @sizeOf(@TypeOf(addr))) catch |err| {
+        log.err("failed to bind netlink socket: {}", .{err});
+        return error.IpLink;
+    };
+
+    // Prepare netlink message to set up loopback interface.
+    const nlmsghdr = linux.nlmsghdr;
+    const ifinfomsg = linux.ifinfomsg;
+
+    const IFF_UP = 0x1;
+
+    var msg_buf: [1024]u8 align(4) = std.mem.zeroes([1024]u8);
+    const hdr = @as(*nlmsghdr, @ptrCast(@alignCast(&msg_buf[0])));
+    const ifi = @as(*ifinfomsg, @ptrCast(@alignCast(&msg_buf[@sizeOf(nlmsghdr)])));
+
+    hdr.* = .{
+        .len = @sizeOf(nlmsghdr) + @sizeOf(ifinfomsg),
+        .type = .RTM_NEWLINK,
+        .flags = linux.NLM_F_REQUEST | linux.NLM_F_ACK,
+        .seq = 0,
+        .pid = 0,
+    };
+
+    ifi.* = .{
+        .family = linux.AF.UNSPEC,
+        .type = 0,
+        // Seems to be the loopback device, not sure how
+        // to find this value the correct way.
+        .index = 1,
+        .flags = IFF_UP,
+        // man pages say use this value.
+        .change = 0xFFFFFFFF,
+    };
+
+    _ = std.posix.send(sock, msg_buf[0..hdr.len], 0) catch |err| {
+        log.err("failed to send netlink message: {}", .{err});
+        return error.IpLink;
+    };
+
+    // Receive acknowledgment
+    var ack_buf: [1024]u8 = undefined;
+    const ack_len = std.posix.recv(sock, &ack_buf, 0) catch |err| {
+        log.err("failed to receive netlink ack: {}", .{err});
+        return error.IpLink;
+    };
+
+    if (ack_len < @sizeOf(nlmsghdr)) {
+        log.err("netlink ack too short", .{});
+        return error.IpLink;
+    }
+
+    const ack_hdr = @as(*const nlmsghdr, @ptrCast(@alignCast(&ack_buf[0])));
+    if (ack_hdr.type == .ERROR) {
+        if (ack_len >= @sizeOf(nlmsghdr) + @sizeOf(i32)) {
+            const errno = @as(*const i32, @ptrCast(@alignCast(&ack_buf[@sizeOf(nlmsghdr)]))).*;
+            if (errno != 0) {
+                log.err("netlink operation failed with errno: {}", .{-errno});
+                return error.IpLink;
+            }
+        }
+    }
+}
+
+fn fork_and_exit(gpa: std.mem.Allocator) !void {
+    var args_ours = try std.process.argsWithAllocator(gpa);
+    defer args_ours.deinit();
```

Comment:
Given that this is linux only, could use `std.os.argv` for arguably simpler slice-based API